lib/cisco_node_utils/vlan.rb

```diff
@@ -100,19 +100,19 @@ def fabricpath_feature_set(fabricpath_set)
     def mode
       result = config_get('vlan', 'mode', @vlan_id)
       return default_mode if result.nil?
-      result.downcase! if result[/FABRICPATH/]
-      result
+      result[0].downcase! if result[0][/FABRICPATH/]
+      result[0]
     end
 
     def mode=(str)
-      if str.empty?
+      if str[0] == default_mode
         config_set('vlan', 'mode', @vlan_id, 'no', '')
       else
-        if 'fabricpath' == str
+        if 'fabricpath' == str[0]
           fabricpath_feature_set(:enabled) unless
             :enabled == fabricpath_feature
         end
-        config_set('vlan', 'mode', @vlan_id, '', str)
+        config_set('vlan', 'mode', @vlan_id, '', str[0])
       end
     end
```

Comment:
Okay, so let's just add a comment to the getter to explain why this is odd; i.e.:

```
     def mode
        result = config_get('vlan', 'mode', @vlan_id)
        return default_mode if result.nil?
+        # Note: The yaml definition for this property uses 'multiple' as a workaround for a bug
+        # in the N7k nxapi code which displays the 'show vlan' output twice.
+       result[0].downcase! if result[0][/FABRICPATH/]
+       result[0]
      end
```

Also, I ran the vlan_mt_full minitest on my n7k (which has an F3 card) and it works, so you're good there.